### PR TITLE
chore: Add tests

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: src/cid_prefix.rs [hunk 1]
```diff
@@ -41,6 +52,37 @@ impl CidPrefix {
         })
     }
 
+    #[allow(dead_code)]
```
Comment (zvolin) [inline src/cid_prefix.rs:55]: maybe `cfg(test)` if you don't want to re-use it in future

File: src/client.rs [hunk 2]
```diff
@@ -350,8 +353,9 @@ where
                         ));
                     }
 
-                    // Blockstore doesn't have the data, add CID in the wantlist. Connection handler will
-                    // read the list and do the request.
+                    // Blockstore doesn't have the data add CID in the wantlist.
```
Comment (zvolin) [inline src/client.rs:356]:
```suggestion
                    // If blockstore doesn't have the data, add CID in the wantlist.
```

File: src/cid_prefix.rs (hunk 1 above)
Comment (oblique) [reply src/cid_prefix.rs:55]: I think it will be used on the server implementation

**Review by fl0rek — APPROVED** [review]

File: src/client.rs [hunk 3]
```diff
@@ -532,3 +535,424 @@ impl<const S: usize> Drop for ClientConnectionHandler<S> {
...
+            let ev = poll_fn(|cx| client.poll(cx)).await;
+
+            let (peer_id, wantlist, send_state) = match ev {
+                ToSwarm::NotifyHandler {
+                    peer_id,
+                    event: ToHandlerEvent::SendWantlist(wantlist, send_state),
+                    ..
+                } => (peer_id, wantlist, send_state),
+                _ => unreachable!(),
+            };
```
[the remark this answers is not in the document]
Comment (zvolin) [reply src/client.rs:617]: or even returning only the peer_id and wantlist

**Review by zvolin — APPROVED** [review]
looks great, this was also a great overview how bitswap works

File: src/client.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (oblique) [reply src/client.rs]: done

**Pull request merged**